File: problems/2640_find_the_score_of_all_prefixes_of_an_array/Solution.java
class Solution {
    public long[] findPrefixScore(int[] nums) {
        long[] ans = new long[nums.length];
        long mx = 0, s = 0;
        for (int i = 0; i < nums.length; i++) {
            mx = Math.max(mx, nums[i]);
            s += nums[i] + mx;
            ans[i] = s;
        }
        return ans;
    }
}

File: problems/2640_find_the_score_of_all_prefixes_of_an_array/solution.cpp
#include <vector>
#include <string>
#include <unordered_map>
#include <unordered_set>
#include <algorithm>
#include <climits>
#include <queue>
#include <stack>
#include <cmath>
#include <numeric>
#include <set>
#include <map>
using namespace std;

class Solution {
public:
    vector<long long> findPrefixScore(vector<int>& nums) {
        vector<long long> ans;
        long long mx = 0, s = 0;
        for (int n : nums) { mx = max(mx, (long long)n); s += n + mx; ans.push_back(s); }
        return ans;
    }
};
